### ISSUE TITLE
The loss function seems to be problematic

### ISSUE THREAD

**Issue by ChuanMeng** [suchanadatta/qppBERT-PL#5]
https://github.com/suchanadatta/qppBERT-PL/blob/d531d1754b96a4c2d2ef5f914843429656b4acc7/qpp_model/train.py#L128

Hi Suchana,
I'm training the model using your script but I find that the model is not being optimised properly and the loss function seems to be problematic. Because in “u_loss += F.nll_loss(utility, torch.tensor([numrel])).item()” the "item" operation would take the loss out of the computation graph, i.e., "item" converts the loss into a plain python number and plain python number can only live on the CPU and it doesn't have gradients, and so the model cannot be optimised according to the gradients.

I modify your code in this way. Do you think it is okay?
```
u_loss= F.nll_loss(utility, torch.tensor([numrel]))
u_loss =u_loss /25
u_loss.backward()

if count == 25: 
    torch.nn.utils.clip_grad_norm_(self.model.parameters(), 1)
    optimizer.step()
    optimizer.zero_grad()
    u_loss=0
    count=0
```
Best wishes,
Chuan Meng

**Comment by suchanadatta** [comment]
Hi Chuan,

Yes, that's correct, many thanks for pointing this issue! 'item()' was used here for some debugging issues back then and pushed mistakenly, it would certainly deviate the loss from the graph. And I think the way you are computing the norm over all gradients together looks fine to me.  

Cheers,
Suchana
